java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -790,6 +791,35 @@ private boolean handleMethodInvocation(Instruction instruction) {
     return false;
   }
 
+  private void enqueueUncheckedExceptions() {
+    programPosition.block.successors()
+        .stream()
+        .map(BytecodeCFG.Block.class::cast)
+        .filter(this::isUncheckedExceptionCatchBlock)
+        .forEach(b -> enqueue(new ProgramPoint(b), stateWithException(programState, b)));
+  }
+
+  private boolean isUncheckedExceptionCatchBlock(BytecodeCFG.Block b) {
+    String exceptionTypeName = b.getExceptionType();
+    if (exceptionTypeName == null) {
+      return false;
+    }
+    Type exceptionType = semanticModel.getClassType(exceptionTypeName);
+    if (exceptionType == null) {
+      return false;
+    }
+    return exceptionType.isSubtypeOf("java.lang.RuntimeException")
```

Comment:
I think I would prefer to have this part shared between both `EGWalker`. In order to be sure that we are handling unchecked exceptions the same way. Maybe by putting it as a `static` method in `ExceptionalSymbolicValue` ?

```
public static boolean isUncheckedException(@Nullable Type exceptionType) {
  if (exceptionType == null) {
    return false;
  }
  return exceptionType.isSubtypeOf("java.lang.RuntimeException")
    || exceptionType.isSubtypeOf("java.lang.Error")
    || exceptionType.is("java.lang.Exception")
    || exceptionType.is("java.lang.Throwable");
}
```